File: コードレビュー/05.マスター/プログラム作成問題：マスター編/20200127_165353689/Wizard.java
import java.util.ArrayList;

import equipment.EquipmentFactory;

// プレイヤー：魔法使い
public class Wizard extends Player {

	// =======================
	// フィールド変数
	// =======================
	Magic[] attackMagic={new Fire(),new Thunder()};
	// =======================
	// コンストラクタ
	// =======================
	public Wizard(String name) {
		super(name);
	}

	// =======================
	// Getter / Setter
	// =======================

	// =======================
	// protected メソッド
	// =======================
	/**
	 * 名前(name)からキャラクターに必要なパラメータを生成する
	 */
	@Override
	protected void MakeCharacter() {
		// 魔術師のパラメータを名前から生成する
		this.defaultHP =(GetNumber(0, 10) + 5) * 10;
		this.defaultMP = GetNumber(1, 50) + 30;
		this.defaultSTR = GetNumber(2, 50) + 1;
		this.defaultDEF = GetNumber(3, 50) + 1;
		this.defaultLUCK = GetNumber(4, 100);
		this.defaultAGI = GetNumber(5, 40) + 20;
		this.paralyze = false;
		this.poison = false;
		this.active=true;

		// 装備の生成
		this.mySword = EquipmentFactory.DecideSword(GetNumber(6,EquipmentFactory.CountSword()));
		this.myArmor = EquipmentFactory.DecideArmor(GetNumber(7,EquipmentFactory.CountArmor()));
		this.myShield = EquipmentFactory.DecideShield(GetNumber(8,EquipmentFactory.CountShield()));
		this.myJewelry = EquipmentFactory.DecideJewelry(GetNumber(9,EquipmentFactory.CountJewelry()));

		this.hp = defaultHP;
		this.mp = defaultMP;
		this.str = defaultSTR + mySword.EquippedSTR(defaultSTR);
		this.def = defaultDEF + myArmor.EquippedDEF(defaultDEF) +myShield.EquippedDEF(defaultDEF);
		this.luck = defaultLUCK;
		this.agi = defaultAGI + myJewelry.EquippedAGI(defaultAGI);

	}

	// ↑を修正して魔法使い用のパラメーターに直す
	// ↓を修正して通常攻撃か魔法攻撃を行うようにする
	/**
	 * {@inheritDoc}<br>
	 * 攻撃側プレイヤー(atacker)のMPが魔法の消費MPに足りている場合は
	 * @param attacker {@inheritDoc}
	 * @param defender {@inheritDoc}
	 */
	@Override
	public void Attack(Player activePlayer,ArrayList<Player> passiveMembers) {

		Player passivePlayer = passiveMembers.get(RandomGenerator.RandomRange(0, passiveMembers.size()));
		//使用する魔法を決定する
		Magic UseMagic = this.attackMagic[RandomGenerator.RandomRange(0,attackMagic.length)];


		if (UseMagic.GetUseMP() <=this.mp) {

			UseMagic.effect(activePlayer,passivePlayer);
			return;

		}
		// 与えるダメージを求める
		System.out.println(activePlayer.GetName() + "　の　攻撃！");
		int damage = activePlayer.CalcDamage(passivePlayer);

		// 求めたダメージを対象プレイヤーに与える
		System.out.println(passivePlayer.GetName() + "　に　" + damage + "　のダメージ！");
		passivePlayer.Damage(damage);

		passivePlayer.Down();

	}

	// =======================
	// private メソッド
	// =======================

	// =======================
	// public メソッド
	// =======================

}

File: コードレビュー/07.Unityアプリ/20200701_USSBB1M39/Assets/Script/Wizard.cs
using System;
using System.Collections;
using System.Collections.Generic;
using UnityEngine;

/// <summary>
/// 職業が魔法使いのプレイヤー
/// </summary>
public class Wizard : Player
{

	// =======================
	// フィールド変数
	// =======================
	Magic[] attackMagic = { new Fire(), new Thunder() };

	/// <summary>
	/// コンストラクタ
	/// </summary>
	/// <param name="name">プレイヤー名</param>
	public Wizard(string name) : base(name)
	{
		this.name = name;
		MakeCharacter();
	}

	// =======================
	// Getter / Setter
	// =======================

	// =======================
	// protected メソッド
	// =======================

	/// <summary>
	/// <inheritdoc/>
	/// </summary>
	protected override void MakeCharacter()
	{
		this.defaultHP = (base.GetNumber(0, 10) + 5) * 10;
		this.defaultMP = base.GetNumber(1, 50) + 30;
		this.hp = base.defaultHP;
		this.mp = base.defaultMP;

		this.str = base.GetNumber(2, 50) + 1;
		this.def = base.GetNumber(3, 50) + 1;
		this.luck = base.GetNumber(4, 100) + 1;
		this.agi = base.GetNumber(5, 40) + 20;

		this.paralyze = false;
		this.poison = false;
		this.active = true;


	}

	// =======================
	// private メソッド
	// =======================

	// =======================
	// public メソッド
	// =======================

	/// <summary>
	/// 攻撃側プレイヤー(activePlayer)のMPが魔法の消費MPに足りている場合は
	/// <inheritdoc/>
	/// </summary>
	/// <param name="activePlayer"><inheritdoc/></param>
	/// <param name="passiveMembers"><inheritdoc/></param>
	public override void Attack(Player activePlayer, List<Player> passiveMembers)
	{
		Debug.Log("Wizard Attack");

		//攻撃対象の決定
		Player passivePlayer = passiveMembers[0];

		if (passiveMembers.Count > 1)
		{
			passivePlayer = passiveMembers[UnityEngine.Random.Range(0, passiveMembers.Count - 1)];

		}

		//使用する魔法を決定する
		Magic UseMagic = this.attackMagic[UnityEngine.Random.Range(0, attackMagic.Length)];

		Debug.Log(string.Format("Wizard Attack {0}", UseMagic.GetName()));
		if (UseMagic.GetUseMP() <= this.mp)
		{
			Debug.Log("Wizard Attack 魔法使えるとき");
			UseMagic.effect(activePlayer, passivePlayer);
			return;

		}
		// 与えるダメージを求める
		LogText.AddLog(string.Format("{0} の 攻撃！", activePlayer.GetName()));
		int damage = activePlayer.CalcDamage(passivePlayer);

		// 求めたダメージを対象プレイヤーに与える
		LogText.AddLog(string.Format("{0} に　{1｝ のダメージ！", passivePlayer.GetName(), damage));
		passivePlayer.Damage(damage);

		passivePlayer.Down();

	}
}
